statsmodels/discrete/discrete_model.py

```diff
@@ -2704,6 +2706,345 @@ def fit_regularized(self, start_params=None, method='l1',
 
         return L1NegativeBinomialResultsWrapper(discretefit)
 
+class NegativeBinomialP(CountModel):
+    __doc__ = """
+    Negative Binomial model for count data
+    %(params)s
+    %(extra_params)s
+    Attributes
+    -----------
+    endog : array
+        A reference to the endogenous response variable
+    exog : array
+        A reference to the exogenous design.
+    """ % {'params' : base._model_params_doc,
+           'extra_params' :
+           """offset : array_like
+        Offset is added to the linear prediction with coefficient equal to 1.
+    exposure : array_like
+        Log(exposure) is added to the linear prediction with coefficient
+        equal to 1.
+    """ + base._missing_param_doc}
+
+    def __init__(self, endog, exog, p=1, offset=None,
+                       exposure=None, missing='none', **kwargs):
+        super(NegativeBinomialP, self).__init__(endog, exog, offset=offset,
+                                                  exposure=exposure,
+                                                  missing=missing, **kwargs)
+        self.parametrization = p
+        self.exog_names.append('alpha')
+        self.k_extra = 1
+        self._transparams = False
+
+    def loglike(self, params):
+        """
+        Loglikelihood of Negative Binomial model
+
+        Parameters
+        ----------
+        params : array-like
+            The parameters of the model.
+        Returns
+        -------
+        loglike : float
+            The log-likelihood function of the model evaluated at `params`.
+            See notes.
+        """
+        return np.sum(self.loglikeobs(params))
+
+    def loglikeobs(self, params):
+        """
+        Loglikelihood for observations of Negative Binomial model
```

Comment:
I think we should add "NB-P" in all docstrings (first line), e.g.
"Loglikelihood for observations of Negative Binomial NB-P model"